### PR TITLE
feat(@angular-devkit/build-optimizer): scrub ɵsetClassMetadata and ɵɵsetNgModuleScope calls

### PR DESCRIPTION
**Pull request by filipesilva**

### CODE REVIEW

**Comment by filipesilva** [comment]
Note to reviewers: `feat(@angular-devkit/build-optimizer): scrub ngFactoryDef and ngComponentDef assignments` is the main commit that, the others are smaller changes.

File: packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts [hunk 1]
```diff
@@ -50,21 +52,20 @@ function scrubFileTransformer(checker: ts.TypeChecker, isAngularCoreFile: boolea
           return ts.forEachChild(node, checkNodeForDecorators);
         }
         const exprStmt = node as ts.ExpressionStatement;
-        if (isDecoratorAssignmentExpression(exprStmt)) {
+        // Do checks that don't need the typechecker first and bail early.
+        if (isCtorParamsAssignmentExpression(exprStmt)
```
Comment (IgorMinar) [inline packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts]: consider swapping the two checks to optimize for the build speed of Ivy code over the legacy ctorParam format.

Comment (filipesilva) [reply packages/angular_devkit/build_optimizer/src/transforms/scrub-file.ts]: Done

**Comment by filipesilva** [comment]
@vikerman suggested adding a e2e with lazy loaded chunk and track it on the size tracker to prevent regressions.

**Comment by IgorMinar** [comment]
@filipesilva sgtm. I was going to add a similar e2e test to the angular repo as well. will this e2e test you intend to write be in the cli repo? I assume so. I guess it doesn't hurt to have this one in two places, but if you were to add the test just to the angular/angular repo then that would be fine as well.

**Comment by filipesilva** [comment]
For comparison, this is a new project with routing, before and after having a lazy route, and before this PR:

- without lazy routes
```
kamik@RED-X1C6 MINGW64 /d/sandbox/latest-project (master)
$ ng build --prod --output-hashing=none --named-chunks=true

chunk {0} runtime-es2015.js (runtime) 2.82 kB [entry] [rendered]
chunk {1} main-es2015.js (main) 345 kB [initial] [rendered]
chunk {2} polyfills-es2015.js (polyfills) 63.5 kB [initial] [rendered]
chunk {3} polyfills-es5-es2015.js (polyfills-es5) 225 kB [initial] [rendered]
chunk {4} styles.css (styles) 0 bytes [initial] [rendered]
Date: 2019-09-25T12:09:41.845Z - Hash: 43f42a4aa8898b102bc5 - Time: 30050ms
Generating ES5 bundles for differential loading...
ES5 bundle generation complete.
```
- with lazy routes
```
kamik@RED-X1C6 MINGW64 /d/sandbox/latest-project (master)
$ ng build --prod --output-hashing=none --named-chunks=true

chunk {0} runtime-es2015.js (runtime) 4.19 kB [entry] [rendered]
chunk {1} lazy-lazy-module-es2015.js (lazy-lazy-module) 1.33 kB  [rendered]
chunk {2} main-es2015.js (main) 521 kB [initial] [rendered]
chunk {3} polyfills-es2015.js (polyfills) 63.5 kB [initial] [rendered]
chunk {4} polyfills-es5-es2015.js (polyfills-es5) 225 kB [initial] [rendered]
chunk {5} styles.css (styles) 0 bytes [initial] [rendered]
Date: 2019-09-25T12:08:15.711Z - Hash: f164c00660845b6906eb - Time: 29757ms
Generating ES5 bundles for differential loading...
ES5 bundle generation complete.
```

Will update this comment with the size change after this PR when the e2e job uploads the artifacts.

**Comment by filipesilva** [comment]
Note to caretaker: it is expected that the `ci/angular: size` check increases in size, as now we're tracking a new project with lazy routes, which also includes the router.

**Pull request merged**